delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/NTripleParser.scala

```diff
@@ -0,0 +1,31 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing
+
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.delta.kernel.Logger
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.RdfError.ParsingError
+import ch.epfl.bluebrain.nexus.delta.rdf.graph.{Graph, NTriples}
+
+object NTripleParser {
+
+  private val logger = Logger[NTripleParser.type]
+
+  def apply(ntriples: NTriples, rootNodeOpt: Option[Iri]): IO[Option[Graph]] =
+    if (ntriples.isEmpty) {
+      // If nothing is returned by the query, we skip
+      IO.none
+    } else {
+      rootNodeOpt match {
+        case Some(rootNode) =>
+          IO.fromEither(Graph(ntriples.copy(rootNode = rootNode))).map { g =>
+            Some(g.replaceRootNode(rootNode))
```

Comment:
these two lines seem to do the same thing - setting the root node of a graph?

Reply:
`replaceRootNode` seems to do some extra operations